Lib/ufo2ft/featureWriters/gdefFeatureWriter.py

```diff
@@ -14,18 +16,25 @@ class GdefFeatureWriter(BaseFeatureWriter):
     tableTag = "GDEF"
 
     def setContext(self, font, feaFile, compiler=None):
-        ctx = super().setContext(font, feaFile, compiler=compiler)
+        ctx = self.context = SimpleNamespace(
+            font=font,
+            feaFile=feaFile,
+            compiler=compiler,
+            todo=set(),
+        )
+        # skip if a GDEF is in the features
+        if ast.findTable(self.context.feaFile, "GDEF") is not None:
+            return ctx
+
         ctx.orderedGlyphSet = self.getOrderedGlyphSet()
         ctx.ligatureCarets = self._getLigatureCarets()
         ctx.openTypeCategories = self.getOpenTypeCategories()
-        ctx.todo = {"GlyphClassDefs", "LigatureCarets"}
+        ctx.todo.update({"GlyphClassDefs", "LigatureCarets"})
```

Comment:
so we're treating {"GlyphClassDefs", "LigatureCarets"} as the "features" that this particular "feature writer" writes, even though technically they are not 4-letter feature tags as in "kern" or "mark".
In this case, the GdefFeatureWriter should contain these two in its `features` class attribute, like the other writers do.
Then in setContext, I'd expect the same treatment as normal features in "skip" mode: i.e. if any is present, don't overwrite; leave in `todo` the ones that we are going to write. This would allow to have only one of the two hand-coded in features.fea and the other one auto-generated, or the user could filter which one to write with the `features` constructor parameter mechanism that already works with existing writers.
Instead of the mere presence of a GDEF block to decide whether to skip, the writer would check that specific GlyphClassDefs or LigatureCarets blocks are present and update the todo accordingly.

Reply:
OK, that's implemented. I added tests as well.